### PR TITLE
improvement(Depot)

### LINKED ISSUE #5713: Improved display of distribution deposits for a deposit
The display of depots of distributions for a depot is very ambiguous for depots which must make distributions to several depots this ambiguity. This issue proposes to list the deposits for a better display

image

### PR DESCRIPTION
**Pull request by lomamech**
- Improve display in Allowed Destination Depots

closes #5713

### CODE REVIEW

**Comment by lomamech** [comment]
image

File: client/src/modules/depots/modals/depot.modal.js [hunk 1]
```diff
@@ -22,10 +25,43 @@ function DepotModalController($state, Depots, Notify, Session, params) {
     vm.depot.parent_uuid = params.parentUuid;
   }
 
+  Depots.read()
+    .then(depots => {
+      vm.depots = depots;
+
+      let index = 0;
```
Comment (jniles) [inline client/src/modules/depots/modals/depot.modal.js]: FYI, you don't need to do this extra index.  The second argument of `Array.forEach()` is an `index` like you made.  So you can do this:

```js
depots.forEach((depot, index) => {

// everything else 

});
```

File: client/src/modules/depots/modals/depot.modal.html [hunk 2]
```diff
@@ -153,20 +153,37 @@
...
+        <div class="col-md-6">
+          <div ng-repeat ="l in DepotModalCtrl.depotLeft">
+            <div class="checkbox">
+              <label>
+                <input type="checkbox" ng-model="l.checked"  ng-true-value="1" ng-false-value="0">
+                <span translate> {{ l.text }} </span>
+              </label>
+            </div>
+          </div>
+        </div>
```
Comment (mbayopanda) [inline client/src/modules/depots/modals/depot.modal.html:178]: If depotLeft and depotRight are just for displaying depots left and right, you could use just one loop such as : 
```html
<div ng-repeat ="r in DepotModalCtrl.depots" class="col-md-6">
  <div class="checkbox">
    <label>
      <input type="checkbox" ng-model="r.checked" ng-true-value="1" ng-false-value="0">
      <span translate> {{ r.text }} </span>
    </label>
  </div>
</div>
```

**Review by mbayopanda — COMMENTED** [review]
Using checkbox and grid for allowed distribution depot is a good idea 👍🏽 it gives to the user much space to interact and it is more readable.

I just made a comment about the use of `depotLeft` and `depotRight`.

Is it still necessary to have `depotLeft` and `depotRight` while we use just one loop with `<div ng-repeat ="r in DepotModalCtrl.depots" class="col-md-6">`

File: client/src/modules/depots/modals/depot.modal.html (hunk 2 above)
Comment (lomamech) [reply client/src/modules/depots/modals/depot.modal.html:178]: image

I tried with this option, the display stays on one column, my idea on concerning the two-column display was just not to have a very long list

Comment (jmcameron) [reply client/src/modules/depots/modals/depot.modal.html:178]: Take a look at the answer to this:  https://stackoverflow.com/questions/25655207/angularjs-how-to-dynamically-split-list-in-multiple-columns

**Comment by lomamech** [comment]
@mbayopanda can you check again

**Comment by lomamech** [comment]
> @mbayopanda can you check again

image
@jmcameron  I found another way to display on 2 columns
image

**Comment by jmcameron** [comment]
> @jmcameron I found another way to display on 2 columns

Your approach looks pretty robust and automatic.  Nice!

File: client/src/modules/depots/modals/depot.modal.html [hunk 3]
```diff
@@ -153,20 +153,38 @@
...
+      </div>
     </div>
 
     <div class="form-group" ng-class="{'has-error' : DepotForm.description.$invalid && DepotForm.$submitted}">
       <label class="control-label" translate>FORM.LABELS.DESCRIPTION</label>
       <textarea
         class="form-control"
         name="description"
-        ng-model="DepotModalCtrl.depot.description"
+        ng-model=" DepotModalCtrl.depot.description"
```
Comment (jmcameron) [inline client/src/modules/depots/modals/depot.modal.html]: Looks like an extra leading space got into this string

**Comment by jmcameron** [comment]
@lomamech How do I get to this display?

**Comment by jniles** [comment]
@jmcameron you'll need to enable restricting distribution depots.  I think it is in stock settings, but it might be in the depot settings.

**Review by jmcameron — APPROVED** [review]
The update looks good and it works fine.   

Minor suggestion: since the depots will be listed in two columns, perhaps the create depot modal should be a bit wider?

**Review by jniles — APPROVED** [review]
LGTM.  bors r+

**Pull request merged**